src/components/SignUp.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import { Actions } from 'react-native-router-flux';
+import firebase from 'firebase';
+import { ScrollView, Text, TextInput, TouchableOpacity } from 'react-native';
+import { Spinner } from './common';
+
+class SignUp extends Component {
+  state = {
+     firstName: '',
+     lastName: '',
+     userName: '',
+     password: '',
+     email: '',
+     favGenres: [],
+     passwordLength: '',
+     passwordShort: true,
+     loading: false
+  }
+
+onButtonPress() {
+  const { email, password, firstName, lastName, userName } = this.state;
+
+  this.setState({ loading: true });
+
+firebase.auth().createUserWithEmailAndPassword(email, password)
+    .then((user) => {
+      firebase.database().ref(`/users/${user.uid}/userinfo`)
+        .push({ firstName, lastName, userName, email });
+    })
+    .then(() => {
+      Actions.pop();
+    });
+}
+
+checkPassword(text) {
+  this.setState({ password: text });
+
+  if (text.length < 6) {
+    this.setState({
+      passwordLength: 'password must be at least 6 characters',
+      passwordShort: true
+    });
+  } else if (text.length < 25 && text.length > 5) {
+    this.setState({
```

Comment:
What is 5? 6? 25?

As the creator of this code, you know exactly what your intent is. As a reader of this code, I haven't a clue what these numbers represent. These are examples of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).

In the `src` folder, create a `constants.js`. In that file, export constants such as `MAXIMUM_PASSWORD_LENGTH` and `MINIMUM_PASSWORD_LENGTH`. Then import those constants into this file and use them.